api_core/google/api_core/protobuf_helpers.py

```diff
@@ -247,3 +249,41 @@ def setdefault(msg_or_dict, key, value):
     """
     if not get(msg_or_dict, key, default=None):
         set(msg_or_dict, key, value)
+
+
+def fieldmask(original, modified):
+    """Create a field mask by comparing two messages.
+
+    Args:
+        original (~google.protobuf.message.Message): the original message. 
+        modified (~google.protobuf.message.Message): the modified message.
+
+    Returns:
+        google.protobuf.field_mask_pb2.FieldMask: returns a FieldMask instance
+            representing the differences between between two messages.
+
+    Raises:
+        ValueError: If the ``original`` or ``modified`` are not the same type.
+    """
+    if type(original) != type(modified):
+        raise ValueError('fieldmask() expects parameters must be of the same type.')
```

Comment:
Suggestion:

```
ValueError('expected that both original and modified should be of the same type, received "{!r}" and "{!r}".'.format(type(original), type(modified))
```

Always useful to surface the problematic parameter, and no need to include the method name as it's in the stacktrace.

Reply:
Done!